packages/context/src/binding.ts

```diff
@@ -13,6 +13,78 @@ export type BoundValue = any;
 
 export type ValueOrPromise<T> = T | Promise<T>;
 
+/**
+ * Scope for binding values
+ */
+export enum BindingScope {
+  /**
+   * The binding provides a value that is calculated each time. This will be
+   * the default scope if not set.
+   *
+   * For example, with the following context hierarchy:
+   *
+   * - c0 (with a binding 'b1' that produces sequential values 0, 1, ...)
+   *   - c1
+   *   - c2
```

Comment:
Perhaps `app` and `req1`, `req2` would make it easier to understand this example in the context of a real application?